source/common/common/hash.cc

```diff
@@ -34,4 +34,40 @@ uint64_t MurmurHash::murmurHash2_64(absl::string_view key, uint64_t seed) {
   return hash;
 }
 
+CharStarSet::~CharStarSet() {
+  std::vector<char*> keys;
+  keys.reserve(hash_set_.size());
+  for (char* p : hash_set_) {
+    keys.push_back(p);
+  }
+  hash_set_.clear();
+  for (char* p : keys) {
+    delete[] p;
+  }
+}
+
+const char* CharStarSet::insert(absl::string_view str) {
+  char* p = new char[str.size() + 1];
+  memcpy(p, str.data(), str.size());
+  p[str.size()] = '\0';
```

Comment:
If null-termination of strings is a common enough operation, could this go in a util library somewhere? common StrUtil?